src/main/java/org/wildfly/security/auth/realm/JaasSecurityRealm.java

```diff
@@ -178,6 +180,9 @@ public boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableEx
                 final Subject subject = new Subject();
                 final LoginContext context  = createLoginContext(loginConfiguration, subject, callbackHandler);
 
+                log.tracef("Trying to authenticate subject %s using LoginContext %s using JaasSecurityRealm",
+                        (principal != null) ? principal.getName() : "null", context.toString());
```

Comment:
This null check isn't really necessary.  Principal should have a reasonable toString(), and tracef can handle null arguments.  Also just use "context" not "context.toString()" - this way, if trace isn't enabled, toString() is never called.